### PR TITLE
feat: implement folder indexer

### PR DESCRIPTION
**Pull request by appflowy**
#### PR Checklist

- [ ] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/rust-lib/flowy-sqlite/src/search.rs [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    let database = Database::new(path, DB_NAME, pool_config).unwrap();
+    let mut conn = database.get_connection().unwrap();
+    (*conn).run_pending_migrations(MIGRATIONS).unwrap();
+    run_migrations(&mut conn).unwrap();
+
+    (tempdir, database)
+  }
+
+  #[test]
+  fn test_migration() {
```
Comment (appflowy) [inline frontend/rust-lib/flowy-sqlite/src/search.rs:167]: the database is always new. Does it necessary to test it?

File: frontend/rust-lib/flowy-sqlite/src/search.rs [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+      index_type: IndexType::Document.to_string(),
+      view_id: view_id.to_owned(),
+      id: page_id.to_owned(),
+      data: text.to_owned(),
+    }
+  }
+}
+
+/// Add search data for searching.
+pub fn add(conn: &mut SqliteConnection, data: &SearchData) -> QueryResult<usize> {
```
Comment (appflowy) [inline frontend/rust-lib/flowy-sqlite/src/search.rs:93]: Can we use the tokio::task::spawn_blocking if there are too many add called

File: frontend/rust-lib/flowy-sqlite/src/search.rs (hunk 1 above)
Comment (Jiraffe7) [reply frontend/rust-lib/flowy-sqlite/src/search.rs:167]: This test was used as part of the dev process. It's not needed per se, so I have removed it.

File: frontend/rust-lib/flowy-sqlite/src/search.rs (hunk 2 above)
Comment (Jiraffe7) [reply frontend/rust-lib/flowy-sqlite/src/search.rs:93]: Do you mean wrap the calls to add in `spawn_blocking`? Currently, the runs in a task spawned by `af_spawn` in `folder_indexer.rs`, so the calls can be wrapped there.

**Pull request closed without merging**